scripts/clang-tidy-diff.sh

```diff
@@ -24,7 +24,7 @@ function realpath() {
 
 CLANG_DIR=$(dirname $(dirname $(realpath $CLANG_TIDY)))
 CLANG_TIDY_DIFF=$CLANG_DIR/share/clang/clang-tidy-diff.py
-ARG="-quiet -p1 -regex=src/.*"
+ARG="-quiet -p1 -iregex=src/.*\.(cpp|cc|c\+\+|cxx|c|cl|h|hpp|m|mm|inc)"
```

Comment:
Does the `\` need to be escaped?

Reply:
Why?

Reply:
I may misunderstand bash, but doesn't it interpret backslash interpreted as an escape character, so clang-tidy will see `-iregex=src/.*.(cpp|...`? This happens to work, but I think it would also include files like `src/not_a_real_cpp`.

Reply:
I see. I checked and it seems it's fed to clang-tidy-diff.py without disappearing, and I also tested with src/not_a_real_cpp and it didn't show up.

Reply:
Thanks for checking!

Reply:
Oh yes, I misunderstood bash. From the reference manual:

> The backslash retains its special meaning only when followed by one of the following characters: ‘$’, ‘`’, ‘"’, ‘\’, or newline. Within double quotes, backslashes that are followed by one of these characters are removed. Backslashes preceding characters without a special meaning are left unmodified.